aten/src/ATen/native/Linear.cpp

```diff
@@ -336,36 +336,36 @@ Tensor einsum(std::string equation, TensorList operands) {
   for (std::size_t i = 0; i < operands.size(); ++i) {
     std::vector<int64_t> perm_shape(perm_index, -1);
     std::vector<int64_t> label_dim(TOTAL_LABELS, -1);
-    const std::vector<int> labels = op_labels[i];
     Tensor operand = operands[i];
-    const auto sizes = operand.sizes();
-    std::size_t j = 0;
+    const auto labels = op_labels[i];
+    const auto original_sizes = operand.sizes();
 
-    for (int label : labels) {
+    std::size_t j = 0;
+    for (auto label : labels) {
       if (label == ELLIPSIS) {
-        // Add missing dimensions under ellipsis
-        int64_t num_dim_diff =
-            ell_num_dim - (operand.dim() - labels.size() + 1);
-        for (int64_t k = 0; k < num_dim_diff; ++k) {
+        // Add missing dimensions covered by the ellipsis
+        const int64_t num_missing_dim =
+            ell_num_dim - (original_sizes.size() - labels.size() + 1);
+        for (int64_t k = 0; k < num_missing_dim; ++k) {
           operand = operand.unsqueeze(j);
         }
         for (int64_t k = 0; k < ell_num_dim; ++k) {
           perm_shape[ell_index + k] = j++;
         }
       } else if (label_dim[label] != -1) {
         // Repeated label, take diagonal
-        int64_t dim = label_dim[label];
+        const auto dim = label_dim[label];
         TORCH_CHECK(
-            sizes[j] == sizes[dim],
```

Comment:
This was also wrong as sizes did not update as dimensions were added to operand.